src/test/java/space/obminyashka/items_exchange/controller/AdvertisementControllerTest.java

```diff
@@ -102,4 +110,63 @@ void findPaginatedAsThumbnails_shouldReturnProperQuantityOfAdvertisementsThumbna
+
+        final var validationMessageSize =
+                createValidationMessage("size", existDtoForUpdate.getSize(), "1", "50");
+        final var validationMessageTopic =
+                createValidationMessage("topic", existDtoForUpdate.getTopic(), "3", "70");
+        final var validationMessageDescription =
+                createValidationMessage("description", existDtoForUpdate.getDescription(), "255");
+        final var validationMessageWhishes =
+                createValidationMessage("wishesToExchange", existDtoForUpdate.getWishesToExchange(), "210");
+        final var validationMessageCity =
+                createValidationMessage("location.city", existDtoForUpdate.getLocation().getCity(), "2", "100");
+        final var validationMessageArea =
+                createValidationMessage("location.area", existDtoForUpdate.getLocation().getArea(), "2", "100");
+        final var validationMessageDistrict =
+                createValidationMessage("location.district", existDtoForUpdate.getLocation().getDistrict(), "2", "100");
+
+        MvcResult mvcResult = mockMvc.perform(put("/adv")
+                .content(asJsonString(existDtoForUpdate))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        Assertions.assertAll(
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageArea, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageDistrict, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageCity, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageSize, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageDescription, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageTopic, mvcResult)),
+                () -> assertTrue(isResponseContainsExpectedResponse(validationMessageWhishes, mvcResult))
+        );
+    }
+
+    private String createValidationMessage(String dtoFieldName, String dtoFieldValue, String min, String max) {
+        return MessageSourceUtil.getMessageSource("invalid.size")
+                .replace("${validatedValue}",
+                        "updateAdvertisement.dto." + dtoFieldName + ": " + dtoFieldValue)
+                .replace("{min}", min)
+                .replace("{max}", max);
+    }
+
+    private String createValidationMessage(String dtoFieldName, String dtoFieldValue, String max) {
+        return MessageSourceUtil.getMessageSource("invalid.max-size")
+                .replace("${validatedValue}",
+                        "updateAdvertisement.dto." + dtoFieldName + ": " + dtoFieldValue)
+                .replace("{max}", max);
+    }
```

Comment:
Move it into AdvertisementDtoCreatingUtil class, please

Reply:
ok